src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -24,17 +37,162 @@ using server = ctx_server<service>;
       *rq.req,
       {ppj::serialization_format::schema_registry_v1_json,
-       ppj::serialization_format::schema_registry_json});
+       ppj::serialization_format::schema_registry_json,
+       ppj::serialization_format::none});
     rq.req.reset();
 
     static const std::vector<std::string_view> schemas_types{"AVRO"};
     auto json_rslt = ppj::rjson_serialize(schemas_types);
     rp.rep->write_body("json", json_rslt);
-    rp.mime_type = res_fmt;
     return ss::make_ready_future<server::reply_t>(std::move(rp));
 }
 
+ss::future<server::reply_t>
+get_schemas_ids_id(server::request_t rq, server::reply_t rp) {
+    rp.mime_type = parse::accept_header(
+      *rq.req,
+      {ppj::serialization_format::schema_registry_v1_json,
+       ppj::serialization_format::schema_registry_json,
+       ppj::serialization_format::none});
+
+    auto id = parse::request_param<schema_id>(*rq.req, "id");
+    rq.req.reset();
+    auto schema = rq.service().schema_store().get_schema(id);
+    if (schema.has_error()) {
+        rp.rep = pandaproxy::errored_body(
+          make_error_code(error_code::schema_id_not_found), "Schema not found");
+        co_return rp;
+    }
+    auto json_rslt = ppj::rjson_serialize(
+      get_schemas_ids_id_response{.definition = schema.value().definition});
+    rp.rep->write_body("json", json_rslt);
+    co_return rp;
+}
+
+ss::future<server::reply_t>
+get_subjects(server::request_t rq, server::reply_t rp) {
+    rp.mime_type = parse::accept_header(
+      *rq.req,
+      {ppj::serialization_format::schema_registry_v1_json,
+       ppj::serialization_format::schema_registry_json,
+       ppj::serialization_format::none});
+
+    rq.req.reset();
+
+    auto subjects = rq.service().schema_store().get_subjects();
+    auto json_rslt{json::rjson_serialize(subjects)};
+    rp.rep->write_body("json", json_rslt);
+    co_return rp;
+}
+
+ss::future<server::reply_t>
+get_subject_versions(server::request_t rq, server::reply_t rp) {
+    rp.mime_type = parse::accept_header(
+      *rq.req,
+      {ppj::serialization_format::schema_registry_v1_json,
+       ppj::serialization_format::schema_registry_json,
+       ppj::serialization_format::none});
+
+    auto sub = parse::request_param<subject>(*rq.req, "subject");
+    rq.req.reset();
+
+    auto versions = rq.service().schema_store().get_versions(sub);
+    if (versions.has_error()) {
+        rp.rep = pandaproxy::errored_body(
+          versions.error(), versions.error().message());
+        co_return rp;
+    }
+
+    auto json_rslt{json::rjson_serialize(versions.value())};
+    rp.rep->write_body("json", json_rslt);
+    co_return rp;
+}
+
+ss::future<server::reply_t>
+post_subject_versions(server::request_t rq, server::reply_t rp) {
+    parse::content_type_header(
+      *rq.req,
+      {ppj::serialization_format::schema_registry_v1_json,
+       ppj::serialization_format::schema_registry_json});
+    rp.mime_type = parse::accept_header(
```

Comment:
should we extract this into `parse::mime_accept_well_known_headers(rp)` seems repeated, but also a nit. up to you.